update import in doc.go of staging/src/k8s.io/metrics

bobsongplus opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:
   update import in doc.go of staging/src/k8s.io/metrics
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Only a partial fix for #68231.

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

TinySong commented:
/assign @piosz

TinySong commented:
Can I add a comment for doc.go  in this PR ?

idealhack commented:
/ok-to-test

nikhita commented:
/kind documentation
/sig instrumentation

/assign @DirectXMan12

nikhita commented:
/release-note-none

DirectXMan12 commented:
this needs an actual description of why this is being done, for posterity's sake.

TinySong commented:
/test pull-kubernetes-integration

TinySong commented:
/test pull-kubernetes-verify

nikhita commented on staging/src/k8s.io/metrics/pkg/apis/external_metrics/doc.go:
```diff
@@ -17,4 +17,4 @@ limitations under the License.
 // +groupName=external.metrics.k8s.io
 
-package external_metrics
+package external_metrics // import "k8s.io/metrics/pkg/apis/external_metrics"
```
There needs to be a newline after this line

nikhita commented:
@TinySong can you update the commit message and PR body to say something like this?

```
Add canonical import paths to metrics packages

The comment will ensure that only imports with
this path will compile.This is needed to make sure
that the vanity url k8s.io is used as the import path
instead of github.com.

See https://golang.org/doc/go1.4#canonicalimports
for more details.
```

bobsongplus committed:
Add canonical import paths to metrics packages

The comment will ensure that only imports with
this path will compile.This is needed to make sure
that the vanity url k8s.io is used as the import path
instead of github.com.

See https://golang.org/doc/go1.4#canonicalimports
for more details.

nikhita commented:
/retest

idealhack commented:
/retest

cc @kubernetes/sig-instrumentation-pr-reviews

brancz commented:
/lgtm
/approve

brancz commented:
/assign @DirectXMan12
